lib/opensusebasetest.pm

```diff
@@ -410,10 +410,19 @@ sub wait_boot {
             # harddisk' is above
             send_key_until_needlematch 'inst-bootmenu-boot-harddisk', 'up';
             boot_local_disk;
-            check_screen(['grub2', 'tianocore-mainmenu'], 15)
+
+            my @tags = qw(grub2 tianocore-mainmenu);
+            push @tags, 'encrypted-disk-password-prompt' if (get_var('ENCRYPT'));
+
+            check_screen(\@tags, 15)
               || die 'niether grub2 or tianocore-mainmenu needles found';
             if (match_has_tag('tianocore-mainmenu')) {
                 $self->handle_uefi_boot_disk_workaround();
+                workaround_type_encrypted_passphrase if (check_screen('encrypted-disk-password-prompt', 10));
+                assert_screen('grub2');
+            }
+            elsif (match_has_tag('encrypted-disk-password-prompt')) {
```

Comment:
This can be simplified a bit by just calling `check_screen` on L#421 and replacing `elsif` with `if`:
```
          if (match_has_tag('tianocore-mainmenu')) {
                $self->handle_uefi_boot_disk_workaround();
                # Maybe we can assert_screen here if ENCRYPT is set
                check_screen('encrypted-disk-password-prompt', 10); 
          }
           if (match_has_tag('encrypted-disk-password-prompt')) {
                workaround_type_encrypted_passphrase;
                assert_screen('grub2');
          }
```
We still will have duplication with L#429, but reworking all affected needles will be too cumbersome I guess.